jpo-asn-runtime/src/main/java/us/dot/its/jpo/asn/runtime/annotations/Asn1ParameterizedTypes.java

```diff
@@ -0,0 +1,65 @@
+package us.dot.its.jpo.asn.runtime.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to specify how to deserialize an ASN1 parameterized type, which
+ * is represented by an abstract generic class in Java.  Modeled after the JsonTypeInfo and
+ * JsonSubTypes annotations in Jackson, but adding the ability to specify that the
+ * id field is an integer, not restricted to being a string like in Jackson.
+ * @author Ivan Yourshaw
+ */
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Asn1ParameterizedTypes {
+
+    /**
+     * @return Name of the property used to determine which type to deserialize
+     */
+    String idProperty();
+
+    /**
+     * @return Type of the id property, which may be integer or string
+     */
+    IdType idType();
+
+    /**
+     * @return Name of the value property containing the payload which can be various types
+     * depending on the generic type parameters.
+     */
+    String valueProperty();
+
+    /**
+     * @return Array of value types mapped to ids.
+     */
+    Type[] value();

```

Comment:
question: will we have multiple values here? The rest of the class is structured for either an int _or_ a string.

Reply:
Yes, that is for nesting multiple values in the annotation.  I added a usage example to the javadoc for that class to show what it looks like.